sdl_android_lib/src/com/smartdevicelink/protocol/BinaryFrameHeader.java

```diff
@@ -27,19 +29,26 @@ public static BinaryFrameHeader parseBinaryHeader(byte[] binHeader) {
 		
 		int _jsonSize = BitConverter.intFromByteArray(binHeader, 8);
 		msg.setJsonSize(_jsonSize);
-		
-		if (_jsonSize > 0) {
-			byte[] _jsonData = new byte[_jsonSize];
-			System.arraycopy(binHeader, 12, _jsonData, 0, _jsonSize);
-			msg.setJsonData(_jsonData);
+
+		try {
+			if (_jsonSize > 0) {
+				byte[] _jsonData = new byte[_jsonSize];
+				System.arraycopy(binHeader, 12, _jsonData, 0, _jsonSize);
+				msg.setJsonData(_jsonData);
+			}
+
+			if (binHeader.length - _jsonSize - 12 > 0) {
+				byte[] _bulkData = new byte[binHeader.length - _jsonSize - 12];
+				System.arraycopy(binHeader, 12 + _jsonSize, _bulkData, 0, _bulkData.length);
+				msg.setBulkData(_bulkData);
+			}
+		} catch (OutOfMemoryError e){
+			Log.e("SdlProtocol", "OOM error due to junk data in BinaryFrameHeader");

```

Comment:
A TAG variable should be created for this with value of `BinaryFrameHeader`. Also, Error message should be a little more generic. `Unable to process data to form header` or similar.